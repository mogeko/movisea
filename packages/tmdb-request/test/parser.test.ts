import { parser } from "@/mod.ts";
import { describe, expect, it } from "vitest";

describe("parser", () => {
  it("should parse route", () => {
    const context = parser("/foo/{bar}", {
      headers: { authorization: "xxx" },
      bar: "baz",
    });

    expect(context.method).toEqual("GET");
    expect(context.url).toEqual("https://api.themoviedb.org/3/foo/baz");
    expect(context.headers?.authorization).toEqual("xxx");

    delete context.headers?.["user-agent"];
    expect(context).toMatchSnapshot();
  });

  it("should parse endpoint", () => {
    const context = parser({
      url: "POST /foo/{bar}",
      headers: {
        authorization: "xxx",
        "content-type": "application/json",
      },
      bar: "baz",
    });

    expect(context.method).toEqual("POST");
    expect(context.url).toEqual("https://api.themoviedb.org/3/foo/baz");
    expect(context.headers?.authorization).toEqual("xxx");
    expect(context.headers?.["content-type"]).toEqual("application/json");

    delete context.headers?.["user-agent"];
    expect(context).toMatchSnapshot();
  });

  it("should be covered by route", () => {
    expect(
      parser("/foo/bar", { url: "This should be covered by route" }).url
    ).toEqual("https://api.themoviedb.org/3/foo/bar");
  });
});
