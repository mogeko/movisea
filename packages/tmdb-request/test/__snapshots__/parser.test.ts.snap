// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`parser > should parse endpoint 1`] = `
{
  "baseUrl": "https://api.themoviedb.org/3",
  "headers": {
    "accept": "application/json",
    "authorization": "xxx",
    "content-type": "application/json",
  },
  "method": "POST",
  "url": "/foo/baz",
}
`;

exports[`parser > should parse route 1`] = `
{
  "baseUrl": "https://api.themoviedb.org/3",
  "headers": {
    "accept": "application/json",
    "authorization": "xxx",
  },
  "method": "GET",
  "url": "/foo/baz",
}
`;
